packages/core/src/routes/interaction/types/transformer.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  usernamePasswordPayloadGuard,
+  emailPasswordPayloadGuard,
+  phonePasswordPayloadGuard,
+  emailPasscodePayloadGuard,
+  phonePasscodePayloadGuard,
+  socialConnectorPayloadGuard,
+} from '@logto/schemas';
+import { z } from 'zod';
+
+import type { IdentifierPayload } from './index.js';
+
+export const usernamePasswordPayloadTransformer =
+  usernamePasswordPayloadGuard.transform<IdentifierPayload>(({ username, password }) => ({
+    identity: {
+      type: 'username',
+      value: username,
+    },
+    verification: {
+      type: 'password',
+      value: password,
```

Comment:
i'm doubting if this is too much, since all payloads have the same structure, shall we just convert to a general payload then parse it? since the only difference between payloads are just the type of `verification.value`

Reply:
@gao-sun updated put a literal type for reference use, see if this is slightly better.

Reply:
updated simplified